src/server.c

```diff
@@ -3250,10 +3229,65 @@ void InitServerLast() {
     server.initial_memory_usage = zmalloc_used_memory();
 }
 
+/* The purpose of this function is to try to "glue" consecutive range
+ * key specs in order to build the legacy (first,last,step) spec
+ * used by the COMMAND command.
+ * By far the most common case is just one range spec (e.g. SET)
+ * but some commands' ranges were split into two or more ranges
+ * in order to have different flags for different keys (e.g. SMOVE,
+ * first key is "read write", second key is "write").
+ *
+ * This functions uses very basic heuristics and is "best effort":
+ * 1. Only commands which have only "range" specs are considered
+ * 2. Only range specs with keystep of 1 are considered
+ * 3. The order of the range specs must be ascending (i.e.
+ *    lastkey of spec[i] == firstkey-1 of spec[i+1])
+ *
+ * Returns C_OK on success*/
+int populateCommandLegacyRangeSpec(struct redisCommand *c) {
+    memset(&c->legacy_range_key_spec, 0, sizeof(c->legacy_range_key_spec));
+    c->legacy_range_key_spec.type = KSPEC_INVALID;
+
+    if (c->keys_specs_num == 0)
+        return C_OK;
+
+    if (c->keys_specs_num == 1 && c->keys_specs[0].type == KSPEC_RANGE) {
+        /* Quick win */
+        c->legacy_range_key_spec = c->keys_specs[0];
+        return C_OK;
+    }
+
+    int firstkey = INT_MAX, lastkey = 0;
+    int prev_lastkey = 0;
+    for (int i = 0; i < c->keys_specs_num; i++) {
+        if (c->keys_specs[i].type != KSPEC_RANGE)
+            break;
```

Comment:
Is it `continue`? The current command definition will not have problems with break, but if there are other definitions before the KSPEC_RANGE definition, it may cause the legacy_range_key_spec data to be inconsistent with the old version.
Like this:
```c
    {"sort",sortCommand,-2,
     "write use-memory @list @set @sortedset @dangerous",
     {{KSPEC_KEYWORD,"write",.u.keyword={"STORE",1,1}},
      {KSPEC_RANGE,"read",.u.range={1,1,1}}},
     sortGetKeys},
```
I wrote a test, if I change the code above.
output:
```
127.0.0.1:6379> command info sort
1) 1) "sort"
   2) (integer) -2
   3) 1) write
      2) denyoom
      3) movablekeys
   4) (integer) 0
   5) (integer) 0
   6) (integer) 0
```

The correct output should be:
```
127.0.0.1:6379> command info sort
1) 1) "sort"
   2) (integer) -2
   3) 1) write
      2) denyoom
      3) movablekeys
   4) (integer) 1
   5) (integer) 1
   6) (integer) 1
```